sympy/geometry/point.py

```diff
@@ -266,7 +264,6 @@ def distance(self, p):
         sqrt(x**2 + y**2)
 
         """
-        p = Point(p)
         return sqrt(sum([(a - b)**2 for a, b in zip(self.args, p.args)]))
```

Comment:
I think you will have to test if p is an instance of Point here, however: to convert float to Rational if given as a tuple and to allow args to be used in the zip.